File: frontend/src/types/domain.ts
export interface Domain {
    id: string;
    name: string;
    only_premium: boolean;
    max_usage: number;
    current_usage: number;
    expires_at: string;
    created_at: string;
    updated_at: string;
}

export interface DomainAssignment {
    id: number;
    uid: number;
    domain_id: string;
    assigned_at: string;
    created_at: string;
    updated_at: string;
}

export interface DomainWithDetails {
    assignment: DomainAssignment;
    domain: Domain;
    is_expiring: boolean;
}

export interface AssignDomainRequest {
    domain_id: string;
}

export interface RenewDomainRequest {
    duration_days: number;
}

export interface DomainSelectionCheckResult {
    has_selected: boolean;
}

File: backend/models/domain.go
package models

import (
	"time"
)

type Domain struct {
	ID           string    `json:"id" gorm:"primaryKey"`
	Name         string    `json:"name" gorm:"unique;not null"`
	OnlyPremium  bool      `json:"only_premium" gorm:"default:false"`
	MaxUsage     int       `json:"max_usage" gorm:"default:0"`
	CurrentUsage int       `json:"current_usage" gorm:"default:0"`
	ExpiresAt    time.Time `json:"expires_at"`
	CreatedAt    time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt    time.Time `json:"updated_at" gorm:"autoUpdateTime"`
}

type DomainAssignment struct {
	ID         uint      `json:"id" gorm:"primaryKey;autoIncrement"`
	UID        uint      `json:"uid" gorm:"not null;index:,unique,composite:idx_uid_domainid"`
	DomainID   string    `json:"domain_id" gorm:"not null;index:,unique,composite:idx_uid_domainid"`
	AssignedAt time.Time `json:"assigned_at" gorm:"not null"`
	CreatedAt  time.Time `json:"created_at" gorm:"autoCreateTime"`
	UpdatedAt  time.Time `json:"updated_at" gorm:"autoUpdateTime"`
}
